src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -169,6 +172,22 @@ internal CommitManager(ImmutableArray<char> potentialCommitCharacters, RecentIte
                 {
                     view.TryMoveCaretToAndEnsureVisible(new SnapshotPoint(subjectBuffer.CurrentSnapshot, change.NewPosition.Value));
                 }
+                else if (editOptions.ComputeMinimalChange)
+                {
+                    // Or, If we're doing a minimal change, then the edit that we make to the 
+                    // buffer may not make the total text change that places the caret where we 
+                    // would expect it to go based on the requested change. In this case, 
+                    // determine where the item should go and set the care manually.
+
+                    // Note: we only want to move the caret if the caret would have been moved 
+                    // by the edit.  i.e. if the caret was actually in the mapped span that 
+                    // we're replacing.
+                    if (view.GetCaretPoint(subjectBuffer) is SnapshotPoint caretPositionInBuffer &&
```

Comment:
is this just to avoid creating a local? or is it actually doing a tyepcheck.

Reply:
it's unclear if the code is actually doing htis part:

>                    // Note: we only want to move the caret if the caret would have been moved 
                    // by the edit.  i.e. if the caret was actually in the mapped span that 
                    // we're replacing.

Reply:
The local is used in the next line in the condition `mappedSpan.IntersectsWith(caretPositionInBuffer)`.

> // by the edit.  i.e. if the caret was actually in the mapped span that 
> // we're replacing.

This is exactly the condition: `mappedSpan.IntersectsWith(caretPositionInBuffer)`.

Reply:
my point is this: you're not actually doing a typecheck.  so why not just assign to a real local?

Reply:
OK. Let us try to extract a local (if you do not like your code I just moved from Controller_Commit)

---
In reply to: [239687119](https://github.com/dotnet/roslyn/pull/31596#discussion_r239687119) [](ancestors = 239687119)

Reply:
i hate my code.  kill it with fire :D